File: medium/python/2971.py
// https://leetcode.com/problems/find-polygon-with-the-largest-perimeter/

class Solution:
    def largestPerimeter(self, nums: List[int]) -> int:
        nums.sort()

        res = -1
        s = sum(nums[:2])

        for e in nums[2:]:
            if e < s:
                res = e + s
            s += e

        return res

File: medium/cpp/2971.cpp
// https://leetcode.com/problems/find-polygon-with-the-largest-perimeter/description/

class Solution
{
public:
    long long largestPerimeter(std::vector<int>& nums)
    {
        std::sort(std::begin(nums), std::end(nums));

        auto res = -1LL;
        auto s = static_cast<long long>(nums[0] + nums[1]);

        for (auto i = 2; i < std::size(nums); i++)
        {
            if (nums[i] < s)
            {
                res = s + nums[i];
            }

            s += nums[i];
        }

        return res;
    }
};